plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -562,7 +563,7 @@ public synchronized void createTable(
             ConnectorSession session,
             Table table,
             PrincipalPrivileges principalPrivileges,
-            Optional<Path> currentPath,
+            Optional<Location> currentPath,
```

Comment:
I believe we want to migrate all usage of  `org.apache.hadoop.fs.Path` eventually. Let me know if I should migrate more in this PR.

Reply:
if we want to maintain compatibility with Hive HMS (or Athena), then per @hashhar 's comment https://github.com/trinodb/trino/pull/17923#pullrequestreview-1482829803 we may need to keep normalization for ever. May require further consideration though

cc @electrum @alexjo2144 @findinpath @losipiuk